apps/src/javalab/Playground.js

```diff
@@ -81,17 +91,36 @@ export default class Playground {
   }
 
   addClickableItem(itemData) {
-    if (this.isGameOver) {
-      // can't add new items if the game is over
-      return;
-    }
+    this.addImageHelper(itemData, true);
   }
 
   addImageItem(itemData) {
-    if (this.isGameOver) {
+    this.addImageHelper(itemData, false);
+  }
+
+  addImageHelper(itemData, isClickable) {
+    // ignore request if the game is over or if the item already exists
+    if (this.isGameOver || this.imageItemExists(itemData)) {
       // can't add new items if the game is over
       return;
     }
+    this.imageItemIds.push(itemData.id);
```

Comment:
Would it be possible for us to not have to maintain this list and rely on redux entirely? My concern is that we have two copies of similar data, one in redux and one here so if by any chance they get out of sync we may have issues. Since this isn't a jsx class though, listening for redux state updates might be a bit weird, but curious what you think - if we're able to have a single source of truth for playground data I think that would be ideal?

Reply:
good idea, I'll refactor to use redux instead